src/app/api/core/services/policies.service.ts

```diff
@@ -0,0 +1,30 @@
+import { UserAction, UserRole } from '@api/core/types/user'
+import { Resource } from '@api/core/types/api'
+import { BaseService } from './base.service'
+import APIError from '@api/core/exceptions/api'
+import httpStatus from 'http-status'
+
+/**
+ * Resource-level policies control service for Users (IU + Client)
+ */
+export class PoliciesService extends BaseService {
+  private defaultPolicies: Record<UserRole.Client, Record<Resource, UserAction[]>> = {
+    [UserRole.Client]: {
+      [Resource.Tasks]: ['read'],
+      [Resource.Status]: ['read'],
+    },
+  }
+
+  authorize(action: UserAction, resource: Resource): boolean | void {
```

Comment:
I found this to be a little confusing. Can you explain what this is supposed to be doing and the scenarios? I think a code comment here would be really helpful